### ISSUE TITLE
Display building:levels in the properties of buildings

### ISSUE THREAD

**Issue by Prusaker** [osmandapp/Osmand#4454]
Is it possible to display the value of the tag **building:levels** in the properties of the buildings?

Thanks in advance :)

**Comment by vshcherb** [comment]
Unfortunately it will be hard to implement cause buildings are not clickable (only housenumbers) and in many cases housenumber is not connected to the building polygon.

**Comment by vshcherb** [comment]
Could you please tell what is the use case for this feature?

**Comment by Prusaker** [comment]
I often go around the cities and put the numbers of houses, apartments and entrances. But I do not know if anyone noticed before me the number of floors in the building. The program does not show this. Because of this, you have to put an extra point on each building with the number of floors.

Maybe there is some other possibility to show how many floors in the building?

**Comment by vshcherb** [comment]
Unfortunately it is hard to do out of the box, it is possible to create a special map with mapcreator where all buildings will be part of POI and that information will be displayed as POI but it should be a custom map.

**Comment by Prusaker** [comment]
I think that addinional user map will be superfluous. I'll try to do without the height of buildings. Thanks for the explanation :)

**Comment by TimMagee** [comment]
The number of levels in a building is a very interesting piece of information. While I understand that OsmAnd is currently restricted to outdoor navigation, level information for buildings would a first step toward indoor navigation. 

Another interesting possibility would be to display the value of the "level" key on all POI that have it so that users can see what level a specific POI is on.

**Comment by scaidermern** [comment]
> Another interesting possibility would be to display the value of the "level" key on all POI that have it so that users can see what level a specific POI is on.

The level key is already displayed :)

**Comment by scaidermern** [comment]
How about a different approach. In Configure map -> Details there are already various options for displaying additional details on the map. There could be a new option called something like "3D mapping" that displays building:levels, building:color, roof:levels, roof:shape and roof:color as building labels, replacing their names and addresses. This way you don't have to make buildings clickable. Might be too much text to display, though.

**Comment by TimMagee** [comment]
Or we could do something like maps.me and display a slightly 2.5D view of all buildings to give an idea of their height. This way we wouldn't have extra text that clutters the screen while we would provide the extra information.

**Comment by scaidermern** [comment]
Yes, an alternative rendering style would also be an option although probably more work.

**Comment by kevinmcmurtrie** [comment]
It would be handy if 3D cities like Tokyo could be visualized.   There are typically three or four levels near transit hubs.  OsmAnd renders this as a mess of dotted lines and it's completely incomprehensible while walking.  If it's four levels, there are two middle levels where you can not even guess whether to go up or down.

**Comment by michaelblyons** [comment]
> I often go around the cities and put the numbers of houses, apartments and entrances. But I do not know if anyone noticed before me the number of floors in the building. The program does not show this. Because of this, you have to put an extra point on each building with the number of floors.
> 
> Maybe there is some other possibility to show how many floors in the building?

Have you discovered [StreetComplete](https://github.com/streetcomplete/StreetComplete)? They are specifically targeted to map source data improvement, with UX geared to data improvements you can make from the street.
